cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/azure_onboarder.py

```diff
@@ -0,0 +1,220 @@
+    report_tf_output(code, stdout, stderr)
+    return code != EX_FAILED
+
+
+# TerraformAction
+def action_apply(t: Terraform, tf_vars: TerraformVars) -> bool:
+    code, stdout, stderr = t.apply(skip_plan=True, var=tf_vars)  # skip_plan means auto-approve
+    report_tf_output(code, stdout, stderr)
+    return code != EX_FAILED
+
+
+# TerraformAction
+def action_destroy(t: Terraform, tf_vars: TerraformVars) -> bool:
+    code, stdout, stderr = t.apply(destroy=IsFlagged, skip_plan=True, var=tf_vars)  # auto-approve
+    report_tf_output(code, stdout, stderr)
+    return code != EX_FAILED
+
+
+def report_tf_output(return_code: int, stdout: str, stderr: str) -> None:
+    if stdout:
+        print(stdout.strip())
+
+    if stderr:
+        print(stderr.strip(), file=sys.stderr)
+
+    if return_code == EX_OK:
+        print("Success")
+    elif return_code == EX_FAILED:
+        print("FAILED", file=sys.stderr)
+    else:
+        print("Return code: ", return_code)  # Terraform uses codes > 1 for reporting detailed status
+
+    print()
+
+
+def get_azure_profiles() -> List[AzureProfile]:
+    # configparser.read silently ignores the fact that ini file doesn't exist, so need to check manually
+    if not os.path.exists(PROFILES_FILE):
+        log.fatal("Azure profiles input file '%s' doesn't exists", PROFILES_FILE)
+        sys.exit(EX_FAILED)
+
+    config = configparser.ConfigParser()
+    try:
+        config.read(PROFILES_FILE)
+    except configparser.Error as err:
+        log.fatal("Error reading Azure profiles file '%s'. Error message: '%s'", PROFILES_FILE, str(err))
+        sys.exit(EX_FAILED)
+
+    output_profiles: List[AzureProfile] = []
+    available_profiles: List[str] = [p for p in config.keys() if p != "DEFAULT"]  # skip ini file "DEFAULT" section
+    for profile_name in available_profiles:
+        profile = config[profile_name]
+        try:
+            resource_groups, storage_accounts = get_resource_groups_and_storage_accounts(profile_name, dict(profile))
+            azure_profile = AzureProfile(
+                name=profile_name,
+                subscription_id=profile["subscription_id"],
+                tenant_id=profile["tenant_id"],
+                location=profile["location"],
+                resource_group_names=resource_groups,
+                storage_account_names=storage_accounts,
+                principal_id=profile["principal_id"],
+                principal_secret=profile["principal_secret"],
+            )
+        except KeyError as err:
+            log.warning("Error reading profile data '%s'. Profile skipped. Error message: '%s'", profile_name, str(err))
+            continue
+        output_profiles.append(azure_profile)
+
+    return output_profiles
+
+
+def get_resource_groups_and_storage_accounts(name: str, profile: Dict[str, Any]) -> Tuple[List[str], List[str]]:
+    split_names = lambda names: [name.strip() for name in names.split(",")] if names else []
```

Comment:
PEP 8: E731 do not assign a lambda expression, use a def
See: https://www.python.org/dev/peps/pep-0008/ -> lambda

Reply:
tried converters but turned out not very helpful in this context
updated to "def"

Reply:
Done